src/nvim/event/stream.c

```diff
@@ -16,6 +16,11 @@
 # include "event/stream.c.generated.h"
 #endif
 
+// For compatbility with libuv < 1.19.0 (tested on 1.18.0)
+#ifndef uv_stream_get_write_queue_size
```

Comment:
`uv_stream_get_write_queue_size` isn't a define, it's a function. It seems like this would _always_ be shadowing the actual function name.

A build-time check in CMakeLists.txt which sets our own define would be the proper way to handle this.

Reply:
Could you please help this stupid Pythonista, who never even opened any `CMakeLists.txt` in his `$EDITOR`?

Reply:
you would have to downgrade `find_package(LibUV 1.28.0 REQUIRED)` . 
Then you can use LibUV_VERSION_MINOR https://gitlab.kitware.com/cmake/cmake/commit/e56aa462976f80762712519a4cf653b8c45bf3db to conditionally add your define. 
(I am no cmake expert either so take it with a grain of salt).

Reply:
> you would have to downgrade `find_package(LibUV 1.28.0 REQUIRED)` .

Hmm, not sure how @mcepl was able to build with these changes if we're requiring a libuv version >= 1.28 and using APIs that were introduced in 1.28.

Reply:
> Hmm, not sure how @mcepl was able to build with these changes if we're requiring a libuv version >= 1.28 and using APIs that were introduced in 1.28.

To answer my own questions, the 1.28.x API use was reverted in #10488 but not the version requirement in CMakeLists.txt.  Also, our `find_package()` handling for libuv is apparently broken since it didn't fail the build when seeing that libuv 1.18 was installed but 1.28 was requested.

Prior to the use of `uv_gettimeofday()`, we had a comment suggesting our minimum libuv version was 1.12.

I'll work on cleaning up the CMake issues.

Reply:
Thank you.